dnn/src/op_cuda.hpp:448: error: (-11:Unknown error code -11) shape and offset provided can potentially leads to OOB access in function 'update'

##### System information (version)

- OpenCV => 4.4pre
- Operating System / Platform => ubuntu 18.04 LTS
- Compiler => gcc 7.5
- Platform x64

##### Detailed description

running opencv dnn facial recognition model on GPU leads to this error when running the "forward" method from cv2.dnn.readNetFromCaffe:

dnn/src/op_cuda.hpp:448: error: (-11:Unknown error code -11) shape and offset provided can potentially leads to OOB access in function 'update'

but works fine on CPU.

model res10_300x300_ssd_iter_140000.caffemodel.
opencv Compiled from the master with CUDA 11, cudnn8 and TensorRT 7.1.3

--- a/modules/dnn/src/dnn.cpp
+++ b/modules/dnn/src/dnn.cpp
@@ -2972,7 +2972,6 @@ struct Net::Impl : public detail::NetImplBase
                             ld.outputBlobsWrappers[0] = wrap(output);
 #endif
                         std::vector<Range> chrange(output.dims, Range::all());
-
                         int ofs = 0;
                         for( i = 0; i < ninputs; i++ )
                         {
@@ -3000,9 +2999,9 @@ struct Net::Impl : public detail::NetImplBase
                             if (preferableBackend == DNN_BACKEND_CUDA)
                             {
                                 auto cuda_wrapper = wrap(output).dynamicCast<CUDABackendWrapper>();
-                                auto offset = chrange[1].start * (output.size[2] * output.size[3]);
-                                auto shape = MatShape{1, chrange[1].size(), output.size[2], output.size[3]};
-                                cuda_wrapper->update(shape, offset);
+                                auto offset = chrange[axis].start * output_slice.total(axis + 1, output.dims);
+                                auto new_shape = shape(output_slice);
+                                cuda_wrapper->update(new_shape, offset);
                                 inp_i_data->outputBlobsWrappers[pin.oid] = cuda_wrapper.staticCast<BackendWrapper>();
                             }
 #endif

